cmd/libsnap-confine-private/string-utils.c

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2016-2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "string-utils.h"
+
+#include <stdarg.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "utils.h"
+
+bool sc_streq(const char *a, const char *b)
+{
+	if (!a || !b) {
+		return false;
+	}
+
+	size_t alen = strlen(a);
+	size_t blen = strlen(b);
+
+	if (alen != blen) {
+		return false;
+	}
+
+	return strncmp(a, b, alen) == 0;
+}
+
+bool sc_endswith(const char *str, const char *suffix)
+{
+	if (!str || !suffix) {
+		return false;
+	}
+
+	size_t xlen = strlen(suffix);
+	size_t slen = strlen(str);
+
+	if (slen < xlen) {
+		return false;
+	}
+
+	return strncmp(str - xlen + slen, suffix, xlen) == 0;
+}
+
+int must_snprintf(char *str, size_t size, const char *format, ...)
```

Comment:
Yes but I didn't want to make the diff huge (it's a very popular function). I will do that separately.